app/src/main/java/com/animal_villa/RetrofitClientInstance.kt

```diff
@@ -0,0 +1,23 @@
+package com.animal_villa
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+/**
+ * RetrofitClientInstance object bootstrap
+ * Boiler plate code set up
+ */
+object RetrofitClientInstance {
+    private var retrofit: Retrofit?= null
+    private val BASE_URL="https://pkgstore.datahub.io/" // (TEMPORARY) the JSON data to read
+
+    val retrofitInstance :Retrofit?
+        get(){
+            if (retrofit==null){
+                retrofit = Retrofit.Builder()
+                    .baseUrl(BASE_URL)
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .build()
+            }
+            return retrofit
+
+        }
+}
```

Comment:
Added the boilerplate code for retrofitclientInstance as I saw its dependencies in the build.gradle. https://www.journaldev.com/13639/retrofit-android-example-tutorial#:~:text=Retrofit%20is%20type%2Dsafe%20REST,compared%20to%20the%20previous%20versions.